web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ Copyright (C) 2015 Electronic Arts Inc.  All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Electronic Arts, Inc. ("EA") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.ea.orbit.web.provider;
+
+import com.ea.orbit.concurrent.Task;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.WriterInterceptor;
+import javax.ws.rs.ext.WriterInterceptorContext;
+
+import java.io.IOException;
+import java.util.concurrent.CompletionException;
+
+@Provider
+public class TaskProvider implements WriterInterceptor
+{
+    @Override
+    public void aroundWriteTo(final WriterInterceptorContext context) throws IOException, WebApplicationException
+    {
+        final Object entity = context.getEntity();
+
+        if(entity instanceof Task)
+        {
+            Object result = entity;
+
+            try
+            {
+                result = ((Task<?>)entity).join();
+            }
+            catch(CompletionException e)
+            {
+                // We need to get the actual exception and rethrow it
+                throw (RuntimeException) ((e.getCause() != null) ? e.getCause() : e);
+            }
+
+            context.setGenericType(null);
```

Comment:
We need to investigate the consequences of setGenericType(null)
I'd be more inclined to:
```
                        context.setGenericType(entityType == null || GenericTypeReflector.isSuperType(entityType, e.getClass())
                                // the actual entity class is narrower than task type parameter
                                // or the task type parameter is undefined or too generic
                                ? e.getClass()
                                // the entity type contains less information than the type parameter
                                // happens for instance if:
                                //    Task<List<SomeDto>> someMethod() { return Task.from(new ArrayList<>(value)) };
                                // where the entity class erases the type parameter
                                : entityType);
```